app/components/app_activity_log_component.rb

```diff
@@ -302,60 +302,105 @@ def programmes_by_id
 
   def decision_expiration_events
     all_programmes = Programme.all.to_a
+    events = []
 
-    AcademicYear.all.filter_map do |academic_year|
+    AcademicYear.find_each do |academic_year|
```

Comment:
You could use `flat_map` here, so you don't need to append to an array.

```rb
AcademicYear.all.flat_map do |academic_year|
  # ...
end
```

It works like a normally `map`, but the return values from the block are concatenated together.